libcontainer/intelrdt/intelrdt.go

```diff
@@ -637,34 +637,27 @@ func (m *Manager) Set(container *configs.Config) error {
 	// For example, on a two-socket machine, the schema line could be
 	// "MB:0=5000;1=7000" which means 5000 MBps memory bandwidth limit on
 	// socket 0 and 7000 MBps memory bandwidth limit on socket 1.
-	if container.IntelRdt != nil {
+	if r := container.IntelRdt; r != nil {
 		path := m.GetPath()
-		l3CacheSchema := container.IntelRdt.L3CacheSchema
-		memBwSchema := container.IntelRdt.MemBwSchema
 
 		// TODO: verify that l3CacheSchema and/or memBwSchema match the
 		// existing schemata if ClosID has been specified. This is a more
 		// involved than reading the file and doing plain string comparison as
 		// the value written in does not necessarily match what gets read out
 		// (leading zeros, cache id ordering etc).
 
-		// Write a single joint schema string to schemata file
-		if l3CacheSchema != "" && memBwSchema != "" {
-			if err := writeFile(path, "schemata", l3CacheSchema+"\n"+memBwSchema); err != nil {
+		if r.L3CacheSchema != "" {
+			if err := writeFile(path, "schemata", r.L3CacheSchema); err != nil {
 				return err
 			}
 		}
-
-		// Write only L3 cache schema string to schemata file
-		if l3CacheSchema != "" && memBwSchema == "" {
-			if err := writeFile(path, "schemata", l3CacheSchema); err != nil {
+		if r.MemBwSchema != "" {
+			if err := writeFile(path, "schemata", r.MemBwSchema); err != nil {
 				return err
 			}
 		}
-
-		// Write only memory bandwidth schema string to schemata file
-		if l3CacheSchema == "" && memBwSchema != "" {
-			if err := writeFile(path, "schemata", memBwSchema); err != nil {
+		if len(r.Schemata) > 0 {
+			if err := writeFile(path, "schemata", strings.Join(r.Schemata, "\n")); err != nil {
```

Comment:
I wonder if it will be better to rewrite `writeFile` to accept multiple data:
```go
func writeFile(dir, file, data... string) error {
...
    for d := range data {
        if d == "" {
            continue
        }
        err := f.WriteString(d)
        ...
    }
    ...
}
```

and then use it like this here:
```go
err := writeFile(path, "schemata", r.L3CacheSchema, r.MemBwSchema, r.Schemata...)
```

Reply:
True. I did a bit more refactoring and ditched the whole `writeFile()` function. The impl is now inline inside `Set()`

@kolyshkin